### ISSUE TITLE
stoploss on exchange issue

### ISSUE THREAD

**Issue by Prossi79** [freqtrade/freqtrade#1726]
I got again stoploss on exchange issues

thats the part of my config:

    "order_types": {
    "buy": "limit",
    "sell": "limit",
    "stoploss": "market",

The part of my strategy file:

    # This attribute will be overridden if the config file contains "stoploss"
    stoploss = -0.15

    # Optimal ticker interval for the strategy
    ticker_interval = '5m'

    # trailing stoploss
    trailing_stop = True   
  
    trailing_stop_positive = 0.005
    trailing_stop_positive_offset = 0.015
    trailing_only_offset_is_reached = False 

and the log from freqtrade - any idea where the problem could be? 

2019-04-02 10:34:30,775 - freqtrade.exchange - WARNING - Giving up retrying: cancel_order()
2019-04-02 10:34:30,775 - freqtrade.freqtradebot - WARNING - Unable to sell trade: Could not cancel order. Message: binance {"code":-2011,"msg":"Unknown order sent."}
2019-04-02 10:34:31,268 - freqtrade.freqtradebot - INFO - Trailing stoploss: cancelling current stoploss on exchange in order to add another one ...
2019-04-02 10:34:31,771 - freqtrade.exchange - WARNING - cancel_order() returned exception: "Could not cancel order. Message: binance {"code":-2011,"msg":"Unknown order sent."}"
2019-04-02 10:34:31,771 - freqtrade.exchange - WARNING - retrying cancel_order() still for 3 times
2019-04-02 10:34:32,272 - freqtrade.exchange - WARNING - cancel_order() returned exception: "Could not cancel order. Message: binance {"code":-2011,"msg":"Unknown order sent."}"
2019-04-02 10:34:32,273 - freqtrade.exchange - WARNING - retrying cancel_order() still for 2 times
2019-04-02 10:34:32,762 - freqtrade.exchange - WARNING - cancel_order() returned exception: "Could not cancel order. Message: binance {"code":-2011,"msg":"Unknown order sent."}"
2019-04-02 10:34:32,762 - freqtrade.exchange - WARNING - retrying cancel_order() still for 1 times
2019-04-02 10:34:33,268 - freqtrade.exchange - WARNING - cancel_order() returned exception: "Could not cancel order. Message: binance {"code":-2011,"msg":"Unknown order sent."}"
2019-04-02 10:34:33,268 - freqtrade.exchange - WARNING - retrying cancel_order() still for 0 times
2019-04-02 10:34:33,776 - freqtrade.exchange - WARNING - cancel_order() returned exception: "Could not cancel order. Message: binance {"code":-2011,"msg":"Unknown order sent."}"
2019-04-02 10:34:33,776 - freqtrade.exchange - WARNING - Giving up retrying: cancel_order()
2019-04-02 10:34:33,777 - freqtrade.freqtradebot - WARNING - Unable to sell trade: Could not cancel order. Message: binance {"code":-2011,"msg":"Unknown order sent."}
2019-04-02 10:34:33,877 - freqtrade.freqtradebot - INFO - Found no buy signals for whitelisted currencies. Trying again..
2019-04-02 10:34:34,771 - freqtrade.freqtradebot - INFO - Trailing stoploss: cancelling current stoploss on exchange in order to add another one ...
2019-04-02 10:34:35,275 - freqtrade.exchange - WARNING - cancel_order() returned exception: "Could not cancel order. Message: binance {"code":-2011,"msg":"Unknown order sent."}"
2019-04-02 10:34:35,276 - freqtrade.exchange - WARNING - retrying cancel_order() still for 3 times
2019-04-02 10:34:35,775 - freqtrade.exchange - WARNING - cancel_order() returned exception: "Could not cancel order. Message: binance {"code":-2011,"msg":"Unknown order sent."}"
2019-04-02 10:34:35,775 - freqtrade.exchange - WARNING - retrying cancel_order() still for 2 times
2019-04-02 10:34:36,276 - freqtrade.exchange - WARNING - cancel_order() returned exception: "Could not cancel order. Message: binance {"code":-2011,"msg":"Unknown order sent."}"
2019-04-02 10:34:36,277 - freqtrade.exchange - WARNING - retrying cancel_order() still for 1 times
2019-04-02 10:34:36,773 - freqtrade.exchange - WARNING - cancel_order() returned exception: "Could not cancel order. Message: binance {"code":-2011,"msg":"Unknown order sent."}"
2019-04-02 10:34:36,773 - freqtrade.exchange - WARNING - retrying cancel_order() still for 0 times
2019-04-02 10:34:37,274 - freqtrade.exchange - WARNING - cancel_order() returned exception: "Could not cancel order. Message: binance {"code":-2011,"msg":"Unknown order sent."}"
2019-04-02 10:34:37,274 - freqtrade.exchange - WARNING - Giving up retrying: cancel_order()
2019-04-02 10:34:37,275 - freqtrade.freqtradebot - WARNING - Unable to sell trade: Could not cancel order. Message: binance {"code":-2011,"msg":"Unknown order sent."}
2019-04-02 10:34:37,778 - freqtrade.freqtradebot - INFO - Trailing stoploss: cancelling current stoploss on exchange in order to add another one ...
2019-04-02 10:34:38,277 - freqtrade.exchange - WARNING - cancel_order() returned exception: "Could not cancel order. Message: binance {"code":-2011,"msg":"Unknown order sent."}"
2019-04-02 10:34:38,277 - freqtrade.exchange - WARNING - retrying cancel_order() still for 3 times
2019-04-02 10:34:38,773 - freqtrade.exchange - WARNING - cancel_order() returned exception: "Could not cancel order. Message: binance {"code":-2011,"msg":"Unknown order sent."}"
2019-04-02 10:34:38,773 - freqtrade.exchange - WARNING - retrying cancel_order() still for 2 times
2019-04-02 10:34:39,274 - freqtrade.exchange - WARNING - cancel_order() returned exception: "Could not cancel order. Message: binance {"code":-2011,"msg":"Unknown order sent."}"
2019-04-02 10:34:39,274 - freqtrade.exchange - WARNING - retrying cancel_order() still for 1 times
2019-04-02 10:34:39,777 - freqtrade.exchange - WARNING - cancel_order() returned exception: "Could not cancel order. Message: binance {"code":-2011,"msg":"Unknown order sent."}"
2019-04-02 10:34:39,778 - freqtrade.exchange - WARNING - retrying cancel_order() still for 0 times
2019-04-02 10:34:40,283 - freqtrade.exchange - WARNING - cancel_order() returned exception: "Could not cancel order. Message: binance {"code":-2011,"msg":"Unknown order sent."}"
2019-04-02 10:34:40,283 - freqtrade.exchange - WARNING - Giving up retrying: cancel_order()
2019-04-02 10:34:40,283 - freqtrade.freqtradebot - WARNING - Unable to sell trade: Could not cancel order. Message: binance {"code":-2011,"msg":"Unknown order sent."}
2019-04-02 10:34:40,403 - freqtrade.freqtradebot - INFO - Found no buy signals for whitelisted currencies. Trying again..
2019-04-02 10:34:41,282 - freqtrade.freqtradebot - INFO - Trailing stoploss: cancelling current stoploss on exchange in order to add another one ...
2019-04-02 10:34:41,784 - freqtrade.exchange - WARNING - cancel_order() returned exception: "Could not cancel order. Message: binance {"code":-2011,"msg":"Unknown order sent."}"
2019-04-02 10:34:41,784 - freqtrade.exchange - WARNING - retrying cancel_order() still for 3 times
2019-04-02 10:34:42,281 - freqtrade.exchange - WARNING - cancel_order() returned exception: "Could not cancel order. Message: binance {"code":-2011,"msg":"Unknown order sent."}"
2019-04-02 10:34:42,281 - freqtrade.exchange - WARNING - retrying cancel_order() still for 2 times
2019-04-02 10:34:42,785 - freqtrade.exchange - WARNING - cancel_order() returned exception: "Could not cancel order. Message: binance {"code":-2011,"msg":"Unknown order sent."}"
2019-04-02 10:34:42,785 - freqtrade.exchange - WARNING - retrying cancel_order() still for 1 times
2019-04-02 10:34:43,284 - freqtrade.exchange - WARNING - cancel_order() returned exception: "Could not cancel order. Message: binance {"code":-2011,"msg":"Unknown order sent."}"
2019-04-02 10:34:43,285 - freqtrade.exchange - WARNING - retrying cancel_order() still for 0 times
2019-04-02 10:34:43,785 - freqtrade.exchange - WARNING - cancel_order() returned exception: "Could not cancel order. Message: binance {"code":-2011,"msg":"Unknown order sent."}"
2019-04-02 10:34:43,786 - freqtrade.exchange - WARNING - Giving up retrying: cancel_order()
2019-04-02 10:34:43,786 - freqtrade.freqtradebot - WARNING - Unable to sell trade: Could not cancel order. Message: binance {"code":-2011,"msg":"Unknown order sent."}

**Comment by mishaker** [comment]
are you able to fetch the stoploss_order_id column in the DB ? just take a look if it contains an ID or it is Null.

**Comment by Prossi79** [comment]
stoploss order id column is NULL ... I had changed to stoploss off exchange after facing this problem ... reloaded the config and restarted the bot. The bot executed the trailing stop loss then .. I guess thats why it is NULL now.

I have seen thtat the buy order was only filled partially then the bot had problems to adjust the trailing stop loss on exchange - could this be the reason? I had a similar problem recently. In my case the stoploss on exchange works very instable...

**Comment by mishaker** [comment]
Ok, I think I know what the problem is. the order is fulfilled and binance actually returns it doesn't recognize the order instead saying it is already fulfilled.

**Comment by mishaker** [comment]
@xmatthias 
cancel_order in this case should not retry, but CCXT returns DependencyException so it keeps retrying: 

```python
@retrier
    def cancel_order(self, order_id: str, pair: str) -> None:
        if self._config['dry_run']:
            return

        try:
            return self._api.cancel_order(order_id, pair)
        except ccxt.InvalidOrder as e:
            raise DependencyException(
                f'Could not cancel order. Message: {e}')
        except (ccxt.NetworkError, ccxt.ExchangeError) as e:
            raise TemporaryError(
                f'Could not cancel order due to {e.__class__.__name__}. Message: {e}')
        except ccxt.BaseError as e:
            raise OperationalException(e)
```

Any idea how we can work around it? :/

**Comment by mishaker** [comment]
@Prossi79 how many open trades did you have while you had these warnings ?

**Comment by xmatthias** [comment]
And what branch are you working with? we fixed a nasty bug related to amounts on develop yesterday / today - but not on the master-branch yet...

**Comment by Prossi79** [comment]
I am on the master branch - shall I switch to develop branch?

@mishaker - I dont know... where cam I check thos?

**Comment by mishaker** [comment]
yes you should switch to develop

**Comment by Prossi79** [comment]
how can i do that?

**Comment by mishaker** [comment]
This issue can also be related to stoploss canceled but new one cannot be placed. we only check the `closed` status, `canceled` should also be checked:

```
if order['status'] == 'closed':
                    trade.sell_reason = SellType.STOPLOSS_ON_EXCHANGE.value
                    trade.update(order)
                    self.notify_sell(trade)
                    result = True
```
